sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala

```diff
@@ -341,4 +345,17 @@ case class InsertIntoHiveTable(
         isSrcLocal = false)
     }
   }
+
+  /**
+   * Verify if the input partition spec has any empty value.
+   */
+  protected def requireNonEmptyValueInPartitionSpec(specs: Seq[TablePartitionSpec]): Unit = {
+    specs.foreach { s =>
+      if (s.values.exists(_.isEmpty)) {
```

Comment:
should this be done in an analyzer rule? how do we do the same check for file source tables?

Reply:
I move it to `PreprocessTableInsertion` rule.